src/librustc_mir/build/scope.rs

```diff
@@ -428,6 +428,21 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
         self.scopes.last().expect("topmost_scope: no scopes present").extent
     }
 
+    /// Returns the scope that we should use as the lifetime of an
+    /// operand. Basically, an operand must live until it is consumed.
+    /// This is similar to, but not quite the same as, the temporary
+    /// scope (which can be larger or smaller).
```

Comment:
In an expression like `foo(bar(), baz())`, we want the operands holding the result of `bar()` and `baz()` to be destroyed right after `foo` is called, and if calling `baz` panics then `bar` is destroyed first - at least that is the ordering that makes the most sense to me.